src/core/annotation.js

```diff
@@ -3540,28 +3541,34 @@ class FreeTextAnnotation extends MarkupAnnotation {
   constructor(params) {
     super(params);
 
-    this.data.hasOwnCanvas = this.data.noRotate;
+    this.data.hasOwnCanvas = true;
 
     const { xref } = params;
     this.data.annotationType = AnnotationType.FREETEXT;
     this.setDefaultAppearance(params);
     if (!this.appearance && this._isOffscreenCanvasSupported) {
       const strokeAlpha = params.dict.get("CA");
       const fakeUnicodeFont = new FakeUnicodeFont(xref, "sans-serif");
-      const fontData = this.data.defaultAppearanceData;
+      const { fontColor, fontSize } = this.data.defaultAppearanceData;
       this.appearance = fakeUnicodeFont.createAppearance(
         this._contents.str,
         this.rectangle,
         this.rotation,
-        fontData.fontSize || 10,
-        fontData.fontColor,
+        fontSize || 10,
+        fontColor,
         strokeAlpha
       );
+      this.data.fontColor = Array.from(fontColor);
+      this.data.fontSize = fontSize || 10;
```

Comment:
Also, here and below, would it not be much better (for API-consistency) to override the existing `this.data.defaultAppearanceData` data with these new values rather than adding a bunch of new properties?